src/sphinx_autodoc_typehints/attributes_patch.py

```diff
@@ -0,0 +1,102 @@
+from functools import partial
+from optparse import Values
+from typing import Any, Tuple
+from unittest.mock import patch
+
+import sphinx.domains.python
+import sphinx.ext.autodoc
+from docutils.parsers.rst import Parser as RstParser
+from docutils.utils import new_document
+from sphinx.addnodes import desc_signature
+from sphinx.application import Sphinx
+from sphinx.domains.python import PyAttribute
+from sphinx.ext.autodoc import AttributeDocumenter
+
+# Defensively check for the things we want to patch
+_parse_annotation = getattr(sphinx.domains.python, "_parse_annotation", None)
+
+# We want to patch:
+# * sphinx.ext.autodoc.stringify_typehint (in sphinx < 6.1)
+# * sphinx.ext.autodoc.stringify_annotation (in sphinx >= 6.1)
+STRINGIFY_PATCH_TARGET = ""
```

Comment:
You could use none just need to annotation it `str | None`, no?

Reply:
That makes a type error when I try to use it unless I add an extra assert. Empty string is falsey and this makes the type checker happy without adding the assert.